### PR TITLE
Updated go linters and CICD tools

### PR DESCRIPTION
**Pull request by torosent**

### CODE REVIEW

File: Engineering/CodeReviews/Go.md [hunk 1]
```diff
@@ -18,6 +18,44 @@ All developers should run ```gofmt``` in a pre-commit hook to ensure standard fo
...
+        "goimports": {
+            "Command": "goimports -l --local github.com/{your-org}/{your-repo}"
+        }
+    },
+    "Enable": [
+        "deadcode",
+        "gofmt",
+        "gocyclo",
+        "goimports",
+        "golint",
```
Comment (jjcollinge) [inline Engineering/CodeReviews/Go.md]: using `golint` via `gometalinter` contradicts the statement made on line 61. Personally, I **do** run `golint` as part of my CI builds on a number of projects and use `//nolint: golint` to disable false positives. What are your thoughts? remove line 61 or remove `golint` from the `gometalint` linters list?

Comment (torosent) [reply Engineering/CodeReviews/Go.md]: Good catch. I think adding `//nolint: golint` is better than removing it completely. It forces developers to "know" their code.

Comment (cloudbeatsch) [reply Engineering/CodeReviews/Go.md]: @timfpark

Comment (timfpark) [reply Engineering/CodeReviews/Go.md]: There will always be false positives in any major project.  My vote would to be go with adding the //nolint: golint guidance to work around it when it happens but have the default be to lint the project.

Comment (torosent) [reply Engineering/CodeReviews/Go.md]: @timfpark updated https://github.com/Microsoft/code-with-engineering-playbook/pull/12/commits/74326a035115edf4f5f3a03e722ac603ffffa507

File: Engineering/CodeReviews/Go.md [hunk 2]
```diff
@@ -35,4 +78,4 @@ The Go language team maintains a list of common [Code Review Comments](https://g
 1. [ ] Does this code have meaningful [Doc Comments](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments)?
 1. [ ] Does this code have meaningful [Package Comments](https://github.com/golang/go/wiki/CodeReviewComments#package-comments)?
 1. [ ] Does this code use [Contexts](https://github.com/golang/go/wiki/CodeReviewComments#contexts) correctly?
-1. [ ] Do unit tests failure with [meaningful messages](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures)?
\ No newline at end of file
+1. [ ] Do unit tests failure with [meaningful messages](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures)?
```
Comment (timfpark) [inline Engineering/CodeReviews/Go.md]: failure -> fail?

Comment (torosent) [reply Engineering/CodeReviews/Go.md]: I didn't write it but I assume it's "fail". I will fix that

**Pull request merged**